verifier/hash_verifier.go

```diff
@@ -307,7 +314,29 @@ func (v *Verifier) compareHashAndUpdate(challengeId uint64, chainRootHash []byte
 		return err
 	}
 	// update metrics if no err
+	logging.Logger.Infof("challenge succeeded for challengeId: %d, hash mismatched", challengeId)
 	v.metricService.IncVerifiedChallenges()
 	v.metricService.IncChallengeSuccess()
 	return err
 }
+
+func (v *Verifier) isInternalSP(spEndpoint string) bool {
```

Comment:
We can use a map to make it more efficient.

Reply:
this would require the endpoints to match exactly